corehq/apps/reports/views.py

```diff
@@ -1113,6 +1113,17 @@ def _render_report_configs(request, configs, domain, owner_id, couch_user, email
     return response.content.decode("utf-8"), excel_attachments
 
 
+def _get_location_safe_report_content(config, domain, couch_user, lang, attach_excel):
+    report_location_safe = any(["computed_owner_location" in report_filter for report_filter in config.filters.keys()])
+    if (toggles.LOCATION_RESTRICTED_SCHEDULED_REPORTS.enabled(domain)
+            and not couch_user.has_permission(domain, 'access_all_locations')
+            and couch_user.get_location_ids(domain)
+            and not report_location_safe):
```

Comment:
`not report_location_safe` condition can be moved up before the other condition checks to exit early if the condition is met so as to avoid any possible DB lookups in other conditions.

Reply:
https://github.com/dimagi/commcare-hq/pull/32964/commits/981a3ec8f8270adb2e8a33fcfa0f2a61b9242143